### ISSUE TITLE
Support display of lines with multiple alternatives ("halufot")

### ISSUE THREAD

**Issue by ShayAdler** [hasadna/open-bus-map-search#176]
Some lines have multiple alternative for the same src + dst combination. This is reflected by a different line ref.
For example: line 24 of Metropolin from Tel Aviv <-> Ramat HaSharon.

Currently in the FE, the selection dropdown will show a duplication, which is a bit unclear:
<img width="634" alt="image" src="https://github.com/hasadna/open-bus-map-search/assets/61648359/3d487cdf-e869-4212-abff-21ed1cfd9fe7">

While the BE does return an indication for that being an alternative (both because of the duplicated line ref, and also in the display name):
<img width="847" alt="image" src="https://github.com/hasadna/open-bus-map-search/assets/61648359/6e6cc5a5-3987-4180-9f14-0b8a9eb99f65">

This is how they display in "אוטובוס קרוב" ׂ(just for inspiration):
<img width="415" alt="image" src="https://github.com/hasadna/open-bus-map-search/assets/61648359/6b99febf-28a9-42f1-9ba4-bb8c1f8e354c">

**Comment by shootermv** [comment]
is the issue on the `single-line-map` page?

**Comment by ShayAdler** [comment]
Basically for all pages that display this drop down (should be the same component AFAIK)

**Comment by shootermv** [comment]
i understand that you want to solve duplicate routes by `combining` them at client level
am i right?

**Comment by ShayAdler** [comment]
nope, the duplication is by design - there are multiple alternatives for some routes. I want to display it as 2 separate routes, but to make it clearer UX wise.
take a look at the bottom of the issue, I attached a pic to how it looks in another app.

**Comment by ArkadiK94** [comment]
This issue continues issue #134 . Can I close the previous issue? I would like to work on this issue pls assign me.

**Comment by ArkadiK94** [comment]
@ShayAdler @NoamGaash

**Comment by NoamGaash** [comment]
@ArkadiK94 sound great, thank you

**Comment by AdlerShay** [comment]
Thanks @ArkadiK94 🙏
As I mentioned in the issue, this requires a small research (basically just a wider validation on more examples), so lmk of you need any help, and pls doc (here in the issue I guess?) Which method you chose

**Comment by ArkadiK94** [comment]
In my opinion we should combine the route alternatives to one route, because as I understand it, there is not a chance that two buses with same line number and src-dst will be scheduled to same point of time with different alternative stops.
Even if we use route_alternative and write it beside the route label it will be unclear for the user which one to choose and the time results will be misleading because we talk about one line number which have alternative stops in some periods.

The images are for the example about route 24 from Metropoline operator. 
In Metropoline site, the user can choose only one option of line 24 for the Tel Aviv - Ramat Hasharon route (or change direction) and where there is an alternative it appears beside the time that it is happens there.
![image](https://github.com/hasadna/open-bus-map-search/assets/76536506/5464d185-543f-4a28-87e0-b9713dd1d9fb)
![image](https://github.com/hasadna/open-bus-map-search/assets/76536506/9ed94ae3-8593-4c96-be4e-07a3d2810c34)
![image](https://github.com/hasadna/open-bus-map-search/assets/76536506/1f5337e6-f9e3-410c-9ef8-4805efb99347)

In Moovit site, you also have one option of this route and can only change direction (if needed)
![image](https://github.com/hasadna/open-bus-map-search/assets/76536506/ad886352-4de5-417b-8735-1fc122f16b78)
![image](https://github.com/hasadna/open-bus-map-search/assets/76536506/1d0b652e-61b9-4bac-b428-32582feff443)
![image](https://github.com/hasadna/open-bus-map-search/assets/76536506/66157be1-ecf9-45a8-8039-ad77b10c4dbc)

@ShayAdler @AdlerShay What do you think?

**Comment by ArkadiK94** [comment]
Is it something that should be done in the frontend or backend? In my opinion,  it related to the backend but I am not sure.

**Comment by NoamGaash** [comment]
I think it should be done in the frontend, or in a separate BFF project. I believe the API should remain as clean as possible, and reflect the real structure of the raw data

**Comment by ArkadiK94** [comment]
> a separate BFF project

Hmm, The back-end of this project is in python. Will it require the BFF to be also written in python? Could I use nodejs? I am asking from the perspective of the knowledge that I already have.

**Comment by ShayAdler** [comment]
@ArkadiK94 you convinced me too :) I guess that most people who are not super-public-transportation-users are not even aware of the concept of "halufot". What are you saying about an hybrid approach - display all the rides under a single route  like you said, but mark the rides that are a part of a halufa somewhat differently:

זמן יציאה 1
זמן יציאה 2
זמן יציאה 3* (tooltip: "נסיעת חלופה, שים לב לפרטי המסלול")
anyway that's not a must.

Technical-wise, I agree that idealy this should be done in the BE. Changing the existing API is not possible, but maybe adding a new route (a more user friendly request that is not exactly parallel to the gtfs structure in the DB, in the same project)  is an option, thou I think it is an over kill for this feature.

**Comment by ArkadiK94** [comment]
@ShayAdler I agree with your suggestion. About my question, do you want the backend for this feature to be implemented in python as well?

**Comment by ShayAdler** [comment]
@ArkadiK94 so as I said, I agree that ideally we can create a new API route in the BE (in the stride-api project, this is very little python and mostly sql).
But, I think it is not a big deal so I prefer this being done in the FE than not being done at all :)

**Comment by ArkadiK94** [comment]
@ShayAdler Hmm, for now , I will do like you suggested from the start.

Combining the routes to one line of route in front-end is a problematic in my opinion. 
In Gaps page and SinglelineMap page , need to combine times of two or more routes.
In Timeline, need to combine stops and times for each stop because so actual and "halufa" routes has same stops.
To make these combines I will need to fetch all the data of the routes,stops,time before displaying it to the user.

**Comment by ArkadiK94** [comment]
>Options solutions I currently think of:

> - Use the src and dst duplication, of there is more than 1 combination of srcX-dstY, than we can tag alternatives (but without knowing which is the main one)
>  - using the route_long_name
>  - using the route_alternative ("#" vs num, but this requieres deeper verification on more examples)
    a combination of the above

I will do use the route_alternative when # or 0 I will not display anything. When a different number I will display the num (with count) and the text "(__ נסיעה חלופית מס)"

**Comment by ArkadiK94** [comment]
Hmm, sometimes the route_alternative is like 7 or 9 without numbers before it. First I though to make a count so if the route_alternative is 9 but we don't have 1 to 8 so make it 2. However, people who knows the site "אוטובוס קרוב" will find it confusing because the number will not match. So I decided to leave the number as it is.

![image](https://github.com/hasadna/open-bus-map-search/assets/76536506/bf03a985-b08f-4375-aa2a-ac8eb712deb2)

**Comment by ArkadiK94** [comment]
@ShayAdler @NoamGaash This issue was resolved with a "temporary" solution in the front end. Would you like to keep this issue as an option to make the better solution which requires a backend changes?

I unassign myself from this issue.
